src/priceHistory/saga.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  Price,
+  fetchPriceHistoryFailure,
+  fetchPriceHistoryStart,
+  fetchPriceHistorySuccess,
+} from 'src/priceHistory/slice'
+import Logger from 'src/utils/Logger'
+import { ensureError } from 'src/utils/ensureError'
+import { fetchWithTimeout } from 'src/utils/fetchWithTimeout'
+import { safely } from 'src/utils/safely'
+import networkConfig from 'src/web3/networkConfig'
+import { call, put, spawn, takeLatest } from 'typed-redux-saga'
+
+const TAG = 'priceHistory/saga'
+
+/**
+ * Fetches the price history for a token from blockchain-api
+ * @param {string} tokenId - The tokenId to fetch the price history for
+ * @param {number} startTimestamp - The timestamp to start fetching prices from defaults to 30 days ago
+ * @param {number} endTimestamp - The timestamp to stop fetching prices at defaults to now
+ */
+export async function fetchTokenPriceHistory(
+  tokenId: string,
+  startTimestamp?: number,
+  endTimestamp?: number
```

Comment:
Why make these optional? It doesn't appear that this function is ever called without these fields and opens the opportunity for a nasty bug with the function not differentiating between 0 and undefined since they are both falsy

Reply:
Fixed in 17d1cbcb9fc7779f7a75612e15f37227467df39c!